NitroxPatcher/Patches/Dynamic/PDAScanner_Scan_Patch.cs

```diff
@@ -53,6 +63,77 @@ public static void Callback()
             }
         }
 
+        public static readonly int PACKET_SENDING_RATE = 500; // in milliseconds
+
+        // Tuple<LatestProgressTime, LatestPacketSendTime, Entry>
+        public static Dictionary<TechType, Tuple<DateTimeOffset, DateTimeOffset, PDAScanner.Entry>> ThrottlingEntries = new Dictionary<TechType, Tuple<DateTimeOffset, DateTimeOffset, PDAScanner.Entry>>();
+        // Store the thread concerning a certain type
+        public static Dictionary<TechType, Thread> ThrottlingThreads = new Dictionary<TechType, Thread>();
+
+        // After 2 seconds without scanning, a packet will be sent with the latest progress
+        public static void Throttle(TechType techType)
+        {
+            do
+            {
+                Thread.Sleep(2000);
+                Log.Debug($"Throttled {techType} waiting");
+            }
+            while (DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() < ThrottlingEntries[techType].Item1.ToUnixTimeMilliseconds() + 2000);
+            Log.Debug($"Throttled {techType} will now be sent");
+
+            // Don't send the packet if the scan is already finished
+            if (!PDAScanner.ContainsCompleteEntry(techType))
+            {
+                NitroxServiceLocator.LocateService<PDAManagerEntry>().Progress(ThrottlingEntries[techType].Item3);
```

Comment:
As this will be executed fairly often `PDAManagerEntry` should be cached.

Reply:
I'm not really sure to understand what you mean for this one

Reply:
`NitroxServiceLocator.LocateService<PDAManagerEntry>()` takes a bit to execute so it should be cached in a field on this class.

Reply:
Oh ok